### PR TITLE
IF-538: Shifts dialog

### PR DESCRIPTION
**Pull request by jsylupp**

### CODE REVIEW

**Comment by mattmikulina** [comment]
Task linked: [IF-538 Add shifts to factory site](https://app.clickup.com/t/2177764/IF-538)

File: fusionbackend/src/main/java/io/fusion/fusionbackend/model/FactorySite.java [hunk 1]
```diff
@@ -81,6 +87,10 @@
     private Double longitude;
     private String imageKey;
 
+    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
+    @JoinColumn(name = "shift_settings_id")
+    private ShiftSettings shiftSettings;
+
```
Comment (davidlane-iteratec) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/model/FactorySite.java:93]: If ShiftSetting cannot exist without FactorySites I would make bidirectional

File: fusionbackend/src/main/java/io/fusion/fusionbackend/model/enums/DayType.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+public enum DayType {
+    MONDAY,
+    TUESDAY,
+    WEDNESDAY,
+    THURSDAY,
+    FRIDAY,
+    SATURDAY,
+    SUNDAY
+}
```
Comment (davidlane-iteratec) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/model/enums/DayType.java]: Why not https://docs.oracle.com/javase/8/docs/api/java/time/DayOfWeek.html ?

File: fusionbackend/src/main/java/io/fusion/fusionbackend/model/Shift.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+@Setter
+@SuperBuilder
+@NoArgsConstructor
+public class Shift extends BaseEntity {
+    @Column(nullable = false)
+    private String name;
+
+    @Column(nullable = false)
+    private Long startMinutes;
+
```
Comment (davidlane-iteratec) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/model/Shift.java:44]: Just a question, don't change anything yet :) but why didn't you go with just LocalTime? Would be a lot easier to work with / manipulate

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/FactorySiteService.java [hunk 4]
```diff
@@ -34,18 +37,39 @@
...
+            if (factorySite.hasShiftSettings()) {
+                ShiftSettings shiftSettings = shiftSettingsRepository
+                        .findDeepById(factorySite.getShiftSettings().getId())
+                        .orElseThrow();
+                factorySite.setShiftSettings(shiftSettings);
+            }
+
+        }
+
+        return factorySites;
```
Comment (davidlane-iteratec) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/service/FactorySiteService.java:68]: Maybe in a separate method or are shiftSettings always of interest?

Comment (davidlane-iteratec) [reply fusionbackend/src/main/java/io/fusion/fusionbackend/service/FactorySiteService.java:68]: Would also recommend using an @Query with fetch joins on a specialized method in factorySiteRepository

File: fusionfrontend/src/assets/i18n/de.json [hunk 5]
```diff
@@ -42,6 +42,24 @@
...
+        "SUNDAY": "Sonntag"
+      },
+      "DAYS_SHORT": {
+        "MONDAY": "Mo",
+        "TUESDAY": "Di",
+        "WEDNESDAY": "Mi",
+        "THURSDAY": "Do",
+        "FRIDAY": "Fr",
+        "SATURDAY": "Sa",
+        "SUNDAY": "So"
```
Comment (davidlane-iteratec) [inline fusionfrontend/src/assets/i18n/de.json:62]: Is this not already provided by the I18N lib?

File: fusionfrontend/src/assets/i18n/en.json [hunk 6]
```diff
@@ -42,6 +42,24 @@
...
+        "SUNDAY": "Sunday"
+      },
+      "DAYS_SHORT": {
+        "MONDAY": "Mo",
+        "TUESDAY": "Tu",
+        "WEDNESDAY": "We",
+        "THURSDAY": "Th",
+        "FRIDAY": "Fr",
+        "SATURDAY": "Sa",
+        "SUNDAY": "Su"
```
Comment (davidlane-iteratec) [inline fusionfrontend/src/assets/i18n/en.json:62]: Same as above

File: fusionfrontend/src/assets/i18n/fr.json [hunk 7]
```diff
@@ -42,6 +42,24 @@
...
+        "SUNDAY": "Dimanche"
+      },
+      "DAYS_SHORT": {
+        "MONDAY": "Lu",
+        "TUESDAY": "Ma",
+        "WEDNESDAY": "Me",
+        "THURSDAY": "Je",
+        "FRIDAY": "Ve",
+        "SATURDAY": "Sa",
+        "SUNDAY": "Di"
```
Comment (davidlane-iteratec) [inline fusionfrontend/src/assets/i18n/fr.json:62]: Same as above

**Review by davidlane-iteratec — CHANGES_REQUESTED** [review]

File: fusionbackend/src/main/java/io/fusion/fusionbackend/model/enums/DayType.java (hunk 2 above)
Comment (jsylupp) [reply fusionbackend/src/main/java/io/fusion/fusionbackend/model/enums/DayType.java]: valid point ;)

File: fusionfrontend/src/assets/i18n/de.json (hunk 5 above)
Comment (jsylupp) [reply fusionfrontend/src/assets/i18n/de.json:62]: Do you know examples? I am not aware of default translations and was not able to find some quickly on the internet..

File: fusionbackend/src/main/java/io/fusion/fusionbackend/model/Shift.java (hunk 3 above)
Comment (jsylupp) [reply fusionbackend/src/main/java/io/fusion/fusionbackend/model/Shift.java:44]: good question.. Will use it next time :)

File: fusionfrontend/src/assets/i18n/de.json (hunk 5 above)
Comment (davidlane-iteratec) [reply fusionfrontend/src/assets/i18n/de.json:62]: Probably the best approach would be to use https://day.js.org/docs/en/i18n/loading-into-browser (previously moment.js)

Comment (jsylupp) [reply fusionfrontend/src/assets/i18n/de.json:62]: due to limit of time I leave it currently as it is, but added a comment to the link

**Pull request merged**